2025-03-02 17:41:51,103 - ERROR - Internal Server Error: /az/quiz_choice
Traceback (most recent call last):
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such table: quizzes

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\main_project\pdfapp\views.py", line 152, in quiz_choice
    return render(request, 'pdfapp/quiz_choice.html', {
        'quizzes' : quizzes
    })
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\defaulttags.py", line 199, in render
    len_values = len(values)
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
    ~~~~~~~~~~~~~~~^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
        chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size
    )
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\models\sql\compiler.py", line 1574, in execute_sql
    cursor.execute(sql, params)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        sql, params, many=False, executor=self._execute
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such table: quizzes
2025-03-02 17:41:51,110 - ERROR - "GET /az/quiz_choice HTTP/1.1" 500 212558
2025-03-02 17:41:51,747 - ERROR - Internal Server Error: /az/quiz_choice
Traceback (most recent call last):
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such table: quizzes

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\main_project\pdfapp\views.py", line 152, in quiz_choice
    return render(request, 'pdfapp/quiz_choice.html', {
        'quizzes' : quizzes
    })
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\template\defaulttags.py", line 199, in render
    len_values = len(values)
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
    ~~~~~~~~~~~~~~~^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
        chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size
    )
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\models\sql\compiler.py", line 1574, in execute_sql
    cursor.execute(sql, params)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        sql, params, many=False, executor=self._execute
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\hikme\Desktop\Programming Projects\Side Projects\websites-projects\pdf-quiz-maker\myvenv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such table: quizzes
2025-03-02 17:41:51,751 - ERROR - "GET /az/quiz_choice HTTP/1.1" 500 213623
2025-03-02 17:41:52,352 - WARNING - Not Found: /favicon.ico
2025-03-02 17:41:52,352 - WARNING - "GET /favicon.ico HTTP/1.1" 404 2550
